aptos-move/framework/aptos-framework/sources/atomic_bridge.move

```diff
@@ -0,0 +1,1374 @@
+        initialize(aptos_framework);
+        set_counterparty_time_lock_duration(aptos_framework, 1);
+        assert!(counterparty_timelock_duration() == 1, 0);
+    }
+
+    #[test(aptos_framework = @aptos_framework, bad = @0xbad)]
+    #[expected_failure(abort_code = 0x50003, location = 0x1::system_addresses)]
+    /// Tests that an incorrect signer cannot update the initiator time lock
+    fun test_not_able_to_set_initiator_time_lock(aptos_framework: &signer, bad: &signer) acquires BridgeConfig {
+        initialize(aptos_framework);
+        set_initiator_time_lock_duration(bad, 1);
+    }
+
+    #[test(aptos_framework = @aptos_framework, bad = @0xbad)]
+    #[expected_failure(abort_code = 0x50003, location = 0x1::system_addresses)]
+    /// Tests that an incorrect signer cannot update the counterparty time lock
+    fun test_not_able_to_set_counterparty_time_lock(aptos_framework: &signer, bad: &signer) acquires BridgeConfig {
+        initialize(aptos_framework);
+        set_counterparty_time_lock_duration(bad, 1);
+    }
+}
+
+module aptos_framework::atomic_bridge {
+    use std::features;
+    use aptos_framework::aptos_coin::AptosCoin;
+    use aptos_framework::bridge_configuration;
+    use aptos_framework::bridge_store;
+    use aptos_framework::coin;
+    use aptos_framework::coin::{BurnCapability, MintCapability};
+    use aptos_framework::fungible_asset::{BurnRef, MintRef};
+    use aptos_framework::system_addresses;
+    #[test_only]
+    use aptos_framework::account;
+    #[test_only]
+    use aptos_framework::aptos_coin;
+    #[test_only]
+    use aptos_framework::block;
+    #[test_only]
+    use aptos_framework::timestamp;
+
+    friend aptos_framework::atomic_bridge_counterparty;
+    friend aptos_framework::atomic_bridge_initiator;
+    friend aptos_framework::genesis;
+
+    struct AptosCoinBurnCapability has key {
+        burn_cap: BurnCapability<AptosCoin>,
+    }
+
+    struct AptosCoinMintCapability has key {
+        mint_cap: MintCapability<AptosCoin>,
+    }
+
+    struct AptosFABurnCapabilities has key {
+        burn_ref: BurnRef,
+    }
+
+    struct AptosFAMintCapabilities has key {
+        burn_ref: MintRef,
+    }
+
+    /// Initializes the atomic bridge by setting up necessary configurations.
+    ///
+    /// @param aptos_framework The signer representing the Aptos framework.
+    public fun initialize(aptos_framework: &signer) {
+        bridge_configuration::initialize(aptos_framework);
+        bridge_store::initialize(aptos_framework);
+    }
+
+    #[test_only]
+    /// Initializes the atomic bridge for testing purposes, including setting up accounts and timestamps.
+    ///
+    /// @param aptos_framework The signer representing the Aptos framework.
+    public fun initialize_for_test(aptos_framework: &signer) {
+        timestamp::set_time_has_started_for_testing(aptos_framework);
+        account::create_account_for_test(@aptos_framework);
+        block::initialize_for_test(aptos_framework, 1);
```

Comment:
Since we're using timestamp, `block::initialize_for_test` is probably not needed. Tests pass without it.